src/components/ProjectDetail/ProjectReports.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useMemo, useState } from 'react';
+import ReportTable from 'components/ReportTable';
+import './styles/Reports.scss';
+import { getUser } from 'utils/auth';
+import { Role } from 'utils/types';
+import CreateReportForm from 'components/CreateReportForm';
+import { Report } from 'utils/types';
+import { REPORTS } from 'fakeData/reports';
+import Pagination from 'components/Pagination';
+
+interface ReportProps {
+   projectId: number;
+}
+
+function ProjectReports({ projectId }: ReportProps) {
+   const currentUser = getUser();
+   const [showCreateReportForm, setShowCreateReportForm] =
+      useState<boolean>(false);
+
+   // list reports
+   let reports = REPORTS.filter((report) => report.projectId === projectId);
+
+   const [pagination, setPagination] = useState({
+      total: reports.length,
+      page: 1,
+      limit: 10,
+   });
+
+   // pagination
+   const handlePagination = (newPage: number) =>
+      setPagination({ ...pagination, page: newPage });
+
+   reports = useMemo<Report[]>(() => {
+      const { page, limit } = pagination;
+      const startIndex = (page - 1) * limit;
+      const endIndex = page * limit;
+      return reports.slice(startIndex, endIndex);
+   }, [pagination, reports]);
```

Comment:
ủa e, sao k để luôn vào useMemo này. (Mà mấy phần lấy data này để tạm thế này cũng được, sau này khi ghép api nhớ chuyển nó thành 1 hook để gọi api và lấy data).
1 cái nữa là a thấy Report table đang k đồng bộ với tasks table (có 1 border bao ngoài)

```suggestion
   const reports = useMemo<Report[]>(() => {
      const reports = REPORTS.filter((report) => report.projectId === projectId);
      const { page, limit } = pagination;
      const startIndex = (page - 1) * limit;
      const endIndex = page * limit;
      return reports.slice(startIndex, endIndex);
   }, [pagination, REPORTS]);
```

Reply:
Tại cái pagination của em nó cái cái **total:reports.length**để check set **disable/enable** cái nút **next/previous** anh, nên là em phải lấy cái **length** của reports trước , để cái pagination ở dưới không nó bị block scope ,cái này do em viết cái pagination nọ hơi dở tí

Reply:
để em nom sửa lại cái pagination xem ạ

Reply:
ok e, cái này cứ để vậy đi, sau này ghép api mình chuyển dùng hook xong update lại cũng được